src/coap_client_zephyr.c

```diff
@@ -178,6 +204,8 @@ static int golioth_coap_cb(struct golioth_req_rsp *rsp)
     {
         if (req->request_complete_event)
         {
+            *req->status = golioth_err_to_status(rsp->err);
```

Comment:
Do we need to make a similar change for libcoap?

Reply:
Yes, that should be the case. Though, I am not sure yet how the implementation would look like.

Reply:
Ok, let's make the change for both platforms at the same time.

Reply:
This PR was updated to cover all platforms, so please take another look @sam-golioth.